scipy/signal/_lti_conversion.py

```diff
@@ -112,83 +114,102 @@ def tf2ss(num, den):
+        If the dimensions n, q, or p could not be determined or if the shapes are
+        incompatible with each other.
+
+    See Also
+    --------
+    StateSpace: Linear Time Invariant system in state-space form.
+    dlti: Discrete-time linear time invariant system base class.
+    tf2ss: Transfer function to state-space representation.
+    ss2tf: State-space to transfer function.
+    ss2zpk: State-space representation to zero-pole-gain representation.
+    cont2discrete: Transform a continuous to a discrete state-space system.
 
+    Examples
+    --------
+    The following example demonstrates that the passed lists are converted into
+    two-dimensional arrays:
+
+    >>> from scipy.signal import abcd_normalize
+    >>> AA, BB, CC, DD = abcd_normalize(A=[[1, 2], [3, 4]], B=[[-1], [5]],
+    ...                                 C=[[4, 5]], D=2.5)
+    >>> AA.shape, BB.shape, CC.shape, DD.shape
+    ((2, 2), (2, 1), (1, 2), (1, 1))
+
+    In the following, the missing parameter C is assumed to be an array of zeros
+    with shape (1, 2):
+
+    >>> from scipy.signal import abcd_normalize
+    >>> AA, BB, CC, DD = abcd_normalize(A=[[1, 2], [3, 4]], B=[[-1], [5]], D=2.5)
+    >>> AA.shape, BB.shape, CC.shape, DD.shape
+    ((2, 2), (2, 1), (1, 2), (1, 1))
+    >>> CC
+    array([[0., 0.]])
     """
-    A, B, C, D = map(_atleast_2d_or_none, (A, B, C, D))
-
-    MA, NA = _shape_or_none(A)
-    MB, NB = _shape_or_none(B)
-    MC, NC = _shape_or_none(C)
-    MD, ND = _shape_or_none(D)
-
-    p = _choice_not_none(MA, MB, NC)
-    q = _choice_not_none(NB, ND)
-    r = _choice_not_none(MC, MD)
-    if p is None or q is None or r is None:
-        raise ValueError("Not enough information on the system.")
-
-    A, B, C, D = map(_none_to_empty_2d, (A, B, C, D))
-    A = _restore(A, (p, p))
-    B = _restore(B, (p, q))
-    C = _restore(C, (r, p))
-    D = _restore(D, (r, q))
+    if A is None and B is None and C is None:
+        raise ValueError("Dimension n is undefined for parameters A = B = C = None!")
+    if B is None and D is None:
+        raise ValueError("Dimension p is undefined for parameters B = D = None!")
+    if C is None and D is None:
+        raise ValueError("Dimension q is undefined for parameters C = D = None!")
+
+    xp = array_namespace(A, B, C, D)
+    A, B, C, D = (xpx.atleast_nd(xp.asarray(M_), ndim=2) if M_ is not None else
+                  xp.zeros((0, 0)) for M_ in (A, B, C, D))
+
+    n = A.shape[0] or B.shape[0] or C.shape[1] or 0  # try finding non-zero dimensions
+    p = B.shape[1] or D.shape[1] or 0
+    q = C.shape[0] or D.shape[0] or 0
+
+    A = xp.zeros((n, n)) if xp_size(A) == 0 else A  # Create zero-matrices if needed
+    B = xp.zeros((n, p)) if xp_size(B) == 0 else B
+    C = xp.zeros((q, n)) if xp_size(C) == 0 else C
+    D = xp.zeros((q, p)) if xp_size(D) == 0 else D
+
+    if A.shape != (n, n):
+        raise ValueError(f"Parameter A was converted to shape {A.shape} != ({n}, {n})!")
```

Comment:
I find this message slightly confusing, maybe this?
```suggestion
        raise ValueError(f"Parameter A has shape {A.shape} but should be ({n}, {n})!")
```

Reply:
Much better. Thanks, Fixed in c004cbd350e273e54b368a3f048c9ccd6dcdd6ef.